### PR TITLE
Added enhancement for Gin Index not supported queries in analyze-sche…

### PR DESCRIPTION
**Pull request by priyanshi-yb**
…ma (#428)

Added conditions for Unsupported Gin index queries (Support ticket on DB side : https://github.com/yugabyte/yugabyte-db/issues/7850) -
1. Gin index on Multicolumn 
2. Gin index on column with ASC/DESC/HASH clause. 

Added check for btree_gin extension which can be created which means it exists in the Database but is not supported with YB gin indexes (Support ticket on DB side: https://github.com/yugabyte/yugabyte-db/issues/9958).

### CODE REVIEW

File: yb-voyager/cmd/analyzeSchema.go [hunk 1]
```diff
@@ -197,6 +198,30 @@ func reportSummary() {
 		reportStruct.Summary.DBObjects = append(reportStruct.Summary.DBObjects, dbObject)
 	}
 }
+// Checks Whether there is a GIN index
```
Comment (amit-yb) [inline yb-voyager/cmd/analyzeSchema.go:201]: In the comment, can you show examples (with full SQL DDL stmt) of all the scenarios reported by this function?

File: yb-voyager/cmd/analyzeSchema.go [hunk 2]
```diff
@@ -197,6 +198,30 @@ func reportSummary() {
...
+func checkGin(sqlInfoArr []sqlInfo, fpath string) {
+	for _, sqlInfo := range sqlInfoArr {
+		idx := ginRegex.FindStringSubmatch(sqlInfo.stmt)
```
Comment (amit-yb) [inline yb-voyager/cmd/analyzeSchema.go:204]: It is a general practice to name the output of regex methods as `match`.

File: yb-voyager/cmd/analyzeSchema.go [hunk 3]
```diff
@@ -197,6 +198,30 @@ func reportSummary() {
...
+		if idx != nil {
+			columnsFromGin :=  strings.Trim(idx[4], `()`)
+			columnList := strings.Split(columnsFromGin, ",")
+			if len(columnList) > 1 {
```
Comment (amit-yb) [inline yb-voyager/cmd/analyzeSchema.go:208]: Isn't it enough to check for the presence of a comma to conclude that it is an index on multiple columns?

File: yb-voyager/cmd/analyzeSchema.go [hunk 4]
```diff
@@ -197,6 +198,30 @@ func reportSummary() {
...
+					"https://github.com/yugabyte/yugabyte-db/issues/7850", "", "INDEX", idx[2], sqlInfo.formattedStmtStr)
+			} else {
+				if strings.Contains(strings.ToUpper(columnList[0]), "ASC") || strings.Contains(strings.ToUpper(columnList[0]), "DESC") || strings.Contains(strings.ToUpper(columnList[0]), "HASH") {
+					reportCase(fpath, "Schema contains gin index on column with ASC/DESC/HASH Clause which is not supported.",
+						"https://github.com/yugabyte/yugabyte-db/issues/7850", "", "INDEX", idx[2], sqlInfo.formattedStmtStr)
+				}
+			}
+		}
+		if strings.Contains(strings.ToLower(sqlInfo.stmt), "extension btree_gin") {
+			reportCase(fpath, "Schema contains btree_gin extension which is not supported for YB",
```
Comment (amit-yb) [inline yb-voyager/cmd/analyzeSchema.go:219]: Can you define a local anonymous function to factor out common values:

```
reportFn := function(msg string, ghIssueID int, name string) {
    reportCase(fpath, msg, fmt.Sprintf(""https://github.com/yugabyte/yugabyte-db/issues/%d", ghIssueID), "INDEX", name, sqlInfo.formattedStmtStr)
}
```

Comment (priyanshi-yb) [reply yb-voyager/cmd/analyzeSchema.go:219]: @amit-yb should this be considered in the refactoring of analyzeSchema later?

**Review by amit-yb — APPROVED** [review]

**Pull request merged**